### PR TITLE
Base: Add a manpage for `printf` / Userland: Make `man` provide `view_width` to `render_for_terminal()`

### PR DESCRIPTION
**Pull request by alimpfard**
I remembered that I never wrote a manpage for `printf`, so here's a manpage for `printf`.
Also messes with `man` a little bit.

@bugaevc provide nitpicks if you want lol

### CODE REVIEW

File: Base/usr/share/man/man1/printf.md [hunk 1]
```diff
@@ -0,0 +1,52 @@
+## Name
+
+printf - format and print data
+
+## Synopsis
+
+```**sh
+printf FORMAT [ARGUMENT]...
```
Comment (bugaevc) [inline Base/usr/share/man/man1/printf.md:8]: no CAPS please :slightly_smiling_face:

Just `printf <format> [arguments...]`

File: Base/usr/share/man/man1/printf.md [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+```
+
+## Description
+
+`printf` formats _ARGUMENT_(s) according to _FORMAT_ and prints the result to standard output.
```
Comment (bugaevc) [inline Base/usr/share/man/man1/printf.md:13]: Here as well

File: Base/usr/share/man/man1/printf.md [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+_FORMAT_ is similar to the C printf format string, with the following differences:
+- The format specifier `b` (`%b`) is not supported
+- The format specifiers that require a writable pointer (e.g. `n`) are not supported
+- The format specifier `q` (`%q`) has a different behaviour, where it shall print a given string as a quoted string, which is safe to use in shell inputs.
```
Comment (bugaevc) [inline Base/usr/share/man/man1/printf.md:18]: This item is a complete sentence, with a period at the end. Others should probably be, too.

File: Base/usr/share/man/man1/printf.md [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+
+# prints "a0\n", ignoring everything after '\c'
+$ printf '%s%d\n\caaaa' a
+
+# prints "123400", as 'x' is an invalid number, and the missing argument for the last '%d' is treated as zero.
+$ printf '%d%d%d' 1 2 3 4 x
+```
+
+## See also
+
```
Comment (bugaevc) [inline Base/usr/share/man/man1/printf.md:52]: Nothing here? If so, leave out the whe section.

**Comment by bugaevc** [comment]
> @bugaevc provide nitpicks if you want lol

:ok_hand:

File: Base/usr/share/man/man1/printf.md (hunk 4 above)
Comment (alimpfard) [reply Base/usr/share/man/man1/printf.md:52]: Nothing I could think of.
Any ideas, or should I leave it out?

Comment (bugaevc) [reply Base/usr/share/man/man1/printf.md]: * `printf`(3), if we have it
* `echo`(1)

Comment (alimpfard) [reply Base/usr/share/man/man1/printf.md]: we don't have `printf`(3) - maybe we should?
I'll add `echo`, makes sense.

**Pull request merged**